src/Language/PureScript/Sugar/Names/Env.hs

```diff
@@ -60,16 +60,16 @@ data Imports = Imports
   , importedModules :: S.Set ModuleName
   -- |
   -- The names of "virtual" modules that come into existence when "import as"
-  -- is used.
+  -- is used, mapped to the real module being imported.
   --
-  , importedVirtualModules :: S.Set ModuleName
+  , importedVirtualModules :: M.Map ModuleName ModuleName
```

Comment:
Actually, all this code in `Docs.Convert` is worryingly similar to what's already in `Sugar.Names`, I think. How would you feel about instead adding a little extra information to the `Exports` data type, mentioning the first "real" module that declarations come from? Then, in `Docs.Convert`, we could just walk over the `Exports` value. We then wouldn't have to look at any `DeclarationRef` values, or `Imports` values, or anything.